From 561609ab5921b1f5f88d55f5b14baeccde6143e9 Mon Sep 17 00:00:00 2001
From: Hugues Fruchet <hugues.fruchet@foss.st.com>
Date: Wed, 30 Oct 2024 16:14:25 +0100
Subject: [PATCH] media: verisilicon: h1: add RGBA support

Add RGBA support, with alpha channel ignored.

Change-Id: Id458c0b466234ce0753214ef078282930f403bd6
Signed-off-by: Hugues Fruchet <hugues.fruchet@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/415272
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Hugues FRUCHET <hugues.fruchet@st.com>
Reviewed-by: Hugues FRUCHET <hugues.fruchet@st.com>
Domain-Review: Hugues FRUCHET <hugues.fruchet@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../media/platform/verisilicon/hantro_h1.c    |  8 ++++++++
 .../platform/verisilicon/stm32mp25_vpu_hw.c   | 20 +++++++++++++++++++
 2 files changed, 28 insertions(+)

--- a/drivers/media/platform/verisilicon/hantro_h1.c
+++ b/drivers/media/platform/verisilicon/hantro_h1.c
@@ -142,10 +142,14 @@ static inline bool is_rgb16(struct v4l2_
 static inline bool is_rgb32(struct v4l2_pix_format_mplane *fmt)
 {
 	return ((fmt->pixelformat == V4L2_PIX_FMT_XBGR32) ||
+		(fmt->pixelformat == V4L2_PIX_FMT_ABGR32) ||
 		(fmt->pixelformat == V4L2_PIX_FMT_BGR32) ||
 		(fmt->pixelformat == V4L2_PIX_FMT_RGBX32) ||
+		(fmt->pixelformat == V4L2_PIX_FMT_RGBA32) ||
 		(fmt->pixelformat == V4L2_PIX_FMT_BGRX32) ||
+		(fmt->pixelformat == V4L2_PIX_FMT_BGRA32) ||
 		(fmt->pixelformat == V4L2_PIX_FMT_XRGB32) ||
+		(fmt->pixelformat == V4L2_PIX_FMT_ARGB32) ||
 		(fmt->pixelformat == V4L2_PIX_FMT_RGB32));
 };
 
@@ -195,6 +199,7 @@ static inline u32 to_cconv_mask(struct v
 		break;
 
 	case V4L2_PIX_FMT_XBGR32:
+	case V4L2_PIX_FMT_ABGR32:
 	case V4L2_PIX_FMT_BGR32:
 		/*
 		 * <MSB>                      <LSB>
@@ -210,6 +215,7 @@ static inline u32 to_cconv_mask(struct v
 		break;
 
 	case V4L2_PIX_FMT_RGBX32:
+	case V4L2_PIX_FMT_RGBA32:
 		/*
 		 * <MSB>                      <LSB>
 		 * XXXXXXXXBBBBBBBBGGGGGGGGRRRRRRRR
@@ -224,6 +230,7 @@ static inline u32 to_cconv_mask(struct v
 		break;
 
 	case V4L2_PIX_FMT_BGRX32:
+	case V4L2_PIX_FMT_BGRA32:
 		/*
 		 * <MSB>                      <LSB>
 		 * RRRRRRRRGGGGGGGGBBBBBBBBXXXXXXXX
@@ -238,6 +245,7 @@ static inline u32 to_cconv_mask(struct v
 		break;
 
 	case V4L2_PIX_FMT_XRGB32:
+	case V4L2_PIX_FMT_ARGB32:
 	case V4L2_PIX_FMT_RGB32:
 		/*
 		 * <MSB>                      <LSB>
--- a/drivers/media/platform/verisilicon/stm32mp25_vpu_hw.c
+++ b/drivers/media/platform/verisilicon/stm32mp25_vpu_hw.c
@@ -114,6 +114,11 @@ static const struct hantro_fmt stm32mp25
 		.enc_fmt = ROCKCHIP_VPU_ENC_FMT_RGB888,
 	},
 	{
+		.fourcc = V4L2_PIX_FMT_ABGR32,
+		.codec_mode = HANTRO_MODE_NONE,
+		.enc_fmt = ROCKCHIP_VPU_ENC_FMT_RGB888,
+	},
+	{
 		.fourcc = V4L2_PIX_FMT_BGR32,
 		.codec_mode = HANTRO_MODE_NONE,
 		.enc_fmt = ROCKCHIP_VPU_ENC_FMT_RGB888,
@@ -124,15 +129,30 @@ static const struct hantro_fmt stm32mp25
 		.enc_fmt = ROCKCHIP_VPU_ENC_FMT_RGB888,
 	},
 	{
+		.fourcc = V4L2_PIX_FMT_RGBA32,
+		.codec_mode = HANTRO_MODE_NONE,
+		.enc_fmt = ROCKCHIP_VPU_ENC_FMT_RGB888,
+	},
+	{
 		.fourcc = V4L2_PIX_FMT_BGRX32,
 		.codec_mode = HANTRO_MODE_NONE,
 		.enc_fmt = ROCKCHIP_VPU_ENC_FMT_RGB888,
 	},
 	{
+		.fourcc = V4L2_PIX_FMT_BGRA32,
+		.codec_mode = HANTRO_MODE_NONE,
+		.enc_fmt = ROCKCHIP_VPU_ENC_FMT_RGB888,
+	},
+	{
 		.fourcc = V4L2_PIX_FMT_XRGB32,
 		.codec_mode = HANTRO_MODE_NONE,
 		.enc_fmt = ROCKCHIP_VPU_ENC_FMT_RGB888,
 	},
+	{
+		.fourcc = V4L2_PIX_FMT_ARGB32,
+		.codec_mode = HANTRO_MODE_NONE,
+		.enc_fmt = ROCKCHIP_VPU_ENC_FMT_RGB888,
+	},
 	{
 		.fourcc = V4L2_PIX_FMT_RGB32,
 		.codec_mode = HANTRO_MODE_NONE,
